front/components/assistant/VoicePicker.tsx

```diff
@@ -338,24 +358,31 @@ export function VoicePicker({
     }
   };
 
-  // ----- Render ---------------------------------------------------------------
-
-  // Use a dynamic recording icon that reflects the input level when recording.
-  const RecordingIcon = () => <VoiceLevelIcon level={level} />;
-
   return (
-    <Button
-      icon={isRecording ? RecordingIcon : ActionMicIcon}
-      isLoading={isTranscribing}
-      variant="ghost-secondary"
-      size="xs"
-      tooltip={computeTooltip(mode, isRecording, isTranscribing)}
-      disabled={disabled}
-      onPointerDown={handlePointerDown}
-      onPointerUp={handlePointerUp}
-      onPointerLeave={handlePointerLeave}
-      onClick={handlePointerClick}
-    />
+    <>
+      <div
+        className={cn(
+          "duration-600 flex items-center justify-end gap-2 overflow-hidden px-2 transition-all ease-in-out",
+          isRecording ? "w-32 opacity-100" : "w-8 opacity-0"
+        )}
+      >
+        <div className="heading-xs font-mono">{formatTime(elapsedSeconds)}</div>
+        <VoiceLevelDisplay level={level} />
+      </div>
```

Comment:
but the voice level and the time in a shrinkable div next to the mic/stop button